runtime/include/chpltypes.h

```diff
@@ -255,10 +255,39 @@ typedef struct chpl_main_argument_s {
 } chpl_main_argument;
 
 static inline _complex128 _chpl_complex128(_real64 re, _real64 im) {
+#ifdef CMPLX
+  return CMPLX(re, im);
+#else
+#ifndef CHPL_DONT_USE_CMPLX_PTR_ALIASING
```

Comment:
At first, I was considering this to be just for C++, but the CI choked last night when I just had cpp skip over these routines for C++ compilations, making me think it may not be guaranteed that all C compilers we use will have `CMPLX` either.  That said, I did not dig much into why the CI choked (i.e., what compiler(s) it uses) and just started looking for something that was not so potentially non-portable.

Reply:
You made me curious why the CI choked.  Looks like it's still using clang 13...  Could it be that something as recent as clang 13 wouldn't have supported `CMPLX`?  Trying godbolt.org, it seemed the answer is "yes"; but then I updated the menu to clang 17, and it still complains about it.  So could it be that Apple clang supports it (due to different headers?) but x86 clang doesn't?  godbolt doesn't seem to support "Apple clang" as an option that I can see, so that's more difficult to test (but it works on my laptop, which is where I developed this).

I also wanted to point out that an obvious downside to my `#ifndef CMPLX` path is that it's not using an initializer, so may make it more expensive to do the cast (?).  But at this point, I think a mathematically correct and portable solution is better than where we were...

Reply:
Interesting. It might be that the `CMPLX` macro is coming from the system headers rather than from anything connected to clang.

> I also wanted to point out that an obvious downside to my `#ifndef CMPLX` path is that it's not using an initializer, so may make it more expensive to do the cast (?).

I'm doubtful that this will be an issue, but you never know. I'd expect the optimizer to handle it readily.